presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/ExtensionsProvidingCommandFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.product.launcher.cli;
+
+import io.airlift.airline.DefaultCommandFactory;
+import io.prestosql.tests.product.launcher.Extensions;
+
+import java.lang.reflect.Constructor;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class ExtensionsProvidingCommandFactory<T>
```

Comment:
Can we have a regular plugin like in Presto or Tempto?

Reply:
Could be, but i wanted this to be simpler. There is no advantage of using service loader over explicitly provided thing.

in tempto there was a clear advantage -- if we wanted not to use service loader, we would need to refactor a lot of code (maybe including testng code), or pass information over static fields (:scary:)

Reply:
So if you are passing things explicitly why do you need reflection then?

Reply:
> So if you are passing things explicitly why do you need reflection then?

Because airline instantiates the command objects. 
We could fix airline for this. eg instead of binding command class, we could bind command factory directly. 
Tbd later. cc @electrum